src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -211,6 +236,262 @@ public function deleteAction(Request $request, int $productId): Response
+    }
+
+    /**
+     * Toggles product status
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", redirectRoute="admin_products_v2_index")
+     * @DemoRestricted(redirectRoute="admin_products_v2_index")
+     *
+     * @param int $productId
+     *
+     * @return RedirectResponse
+     */
+    public function toggleStatusAction(int $productId): RedirectResponse
+    {
+        try {
+            /** @var ProductForEditing $editableProduct */
+            $editableProduct = $this->getQueryBus()->handle(new GetProductForEditing((int) $productId));
+            $this->getCommandBus()->handle(
+                new UpdateProductStatusCommand((int) $productId, !$editableProduct->isActive())
+            );
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (ManufacturerException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_products_v2_index');
+    }
+
+    /**
+     * Updates product position.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_products_v2_index",
+     *     redirectQueryParamsToKeep={"id_category"},
+     *     message="You do not have permission to edit this."
+     * )
+     * @DemoRestricted(
+     *     redirectRoute="admin_products_v2_index",
+     *     redirectQueryParamsToKeep={"id_category"}
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function updatePositionAction(Request $request): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(
+                new UpdateProductPositionCommand(
+                    $request->request->get('positions'),
+                    $request->query->getInt('id_category')
+                )
+            );
+        } catch (CannotUpdateProductPositionException $e) {
+            $errors = $e->getErrors();
+            $this->flashErrors($errors);
+
+            return $this->redirectToRoute('admin_products_v2_index');
+        }
+        $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
+
+        return $this->redirectToRoute('admin_products_v2_index');
+    }
+
+    /**
+     * Delete products in bulk action.
+     *
+     * @AdminSecurity(
+     *     "is_granted('delete', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_products_v2_index",
+     *     message="You do not have permission to delete this."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function bulkDeleteAction(Request $request): RedirectResponse
```

Comment:
We need to discuss with @MatShir if these bulk actions need to be done via a sequence of ajax requests in a modal

Reply:
This is how it works right now in 1.7.8